src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -163,6 +163,19 @@ fn check_rvalue(
                 _ => check_operand(tcx, mir, operand, span),
             }
         }
+        Rvalue::Cast(CastKind::UnsafeFnPointer, _, _) |
+        Rvalue::Cast(CastKind::ReifyFnPointer, _, _) => Err((
+            span,
+            "function pointer casts are not allowed in const fn".into(),
+        )),
+        Rvalue::Cast(CastKind::ClosureFnPointer, _, _) => Err((
+            span,
+            "closures are not allowed in const fn".into(),
```

Comment:
This is not just having a closure, but then casting it to `fn()` (I think). The message should contain the word "cast" one way or another.

I would have just grouped it with the other two fn ptr casts.